### PR TITLE
mdbook-variables: init at 0.2.4

### PR DESCRIPTION
**Pull request by kraftnix**
https://gitlab.com/tglman/mdbook-variables is a mdbook preprocessor which replaces values in double brackets with variables set in book.toml or environment variables.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/md/mdbook-variables/package.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  };
+
+  cargoHash = "sha256-uw1oWIoKi6qsObI4SkEiHwEj9QoxE9jufu9O+ZKM8DI=";
+
+  meta = with lib; {
+    description = "mdBook preprocessor to replace values with env variables.";
+    mainProgram = "mdbook-variables";
+    homepage = "https://gitlab.com/tglman/mdbook-variables";
+    license = licenses.mpl20;
+    maintainers = with maintainers; [ kraftnix ];
```
Comment (Aleksanaa) [inline pkgs/by-name/md/mdbook-variables/package.nix:24]:
```suggestion
  meta = {
    description = "mdBook preprocessor to replace values with env variables";
    mainProgram = "mdbook-variables";
    homepage = "https://gitlab.com/tglman/mdbook-variables";
    license = lib.licenses.mpl20;
    maintainers = with lib.maintainers; [ kraftnix ];
```

File: pkgs/by-name/md/mdbook-variables/package.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+  src = fetchFromGitLab {
+    owner = "tglman";
+    repo = "mdbook-variables";
+    rev = version;
+    hash = "sha256-whvRCV1g2avKegfQpMgYi+E6ETxT2tQqVS2SWRpAqF8=";
+  };
+
+  cargoHash = "sha256-uw1oWIoKi6qsObI4SkEiHwEj9QoxE9jufu9O+ZKM8DI=";
+
```
Comment (Aleksanaa) [inline pkgs/by-name/md/mdbook-variables/package.nix:18]: Would be better to add https://nixos.org/manual/nixpkgs/unstable/#versioncheckhook

File: maintainers/maintainer-list.nix
Comment (Aleksanaa) [inline maintainers/maintainer-list.nix:1]: This should be separated to commit `maintainers: add kraftnix`

**Comment by Aleksanaa** [comment]
@ofborg build mdbook-variables

File: pkgs/by-name/md/mdbook-variables/package.nix [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  nativeInstallCheckInputs = [
+    versionCheckHook
+  ];
+  doInstallCheck = true;
+
+  meta = {
+    description = "mdBook preprocessor to replace values with env variables.";
```
Comment (yunfachi) [inline pkgs/by-name/md/mdbook-variables/package.nix:26]:
```suggestion
    description = "MdBook preprocessor to replace values with env variables";
```
https://github.com/NixOS/nixpkgs/blob/master/doc/stdenv/meta.chapter.md#description-var-meta-description

Comment (Aleksanaa) [reply pkgs/by-name/md/mdbook-variables/package.nix:26]: Can skip first uppercase because mdBook is a project name

**Review by Aleksanaa — APPROVED** [review]

**Review by yunfachi — APPROVED** [review]

**Pull request merged**